File: array/how-many-numbers-are-smaller-than-the-current-number/how_many_numbers_are_smaller_than_the_current_number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums):
        length = len(nums)
        count = [length-1] * length
        for i in range(0,length):
            for j in range(i+1,length):
                if nums[i] == nums[j]:
                    count[i] = count[i] - 1
                    count[j] = count[j] - 1
                elif nums[i] < nums[j]:
                    count[i] = count[i] - 1
                else:
                    count[j] = count[j] - 1
        return count


s = Solution()

s.smallerNumbersThanCurrent([8,1,2,2,3])

File: array/how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function (nums) {
    // return nums.map(num => nums.filter(n => num > n).length);

    let counts = Array(nums.length).fill(nums.length - 1);
    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            if (nums[i] < nums[j]) {
                counts[i]--;
            } else if (nums[i] === nums[j]) {
                counts[j]--;
                counts[i]--;
            } else {
                counts[j]--;
            }
        }
    }
    return counts;
};


smallerNumbersThanCurrent([8, 1, 2, 2, 3])